R/ggplotly.R

```diff
@@ -217,8 +217,24 @@ gg2list <- function(p){
     }
     
     # This extracts essential info for this geom/layer.
-    traces <- layer2traces(L, df, misc)
-
+    if (L$geom$objname == "smooth") {
```

Comment:
Well actually the approach with smoothRibbon and smoothLine is fine with me, but I just think it will be easier to maintain in the future if we put all the geom-specific code inside of layer2traces (rather than being inside of gg2list as it is now).